src/server/scripts/Commands/cs_cast.cpp

```diff
@@ -194,9 +204,8 @@ class cast_commandscript : public CommandScript
         Unit* target = handler->getSelectedUnit();
```

Comment:
please add a new method "getSelectedUnitOrSelf()" just like there is already "getSelectedPlayerOrSelf()" so return the selected unit or session player